### PR TITLE
Replace `Range::step_by` checking with `Iterator::step_by`

### PR DESCRIPTION
**Pull request by oli-obk**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # Change Log
 All notable changes to this project will be documented in this file.
 
+## 0.0.142
+* Deprecated [`range_step_by_zero`]
+* New lint: [`iterator_step_by_zero`]
+
 ## 0.0.141
```
Comment (mcarton) [inline CHANGELOG.md]: 0.0.141 has not been published yet

File: clippy_lints/src/ranges.rs [hunk 2]
```diff
@@ -53,11 +54,12 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for StepByZero {
 
             // Range with step_by(0).
             if name == "step_by" && args.len() == 2 && has_step_by(cx, &args[0]) && is_integer_literal(&args[1], 0) {
```
Comment (mcarton) [inline clippy_lints/src/ranges.rs]: I wonder why we are not evaluating the constant here rather than linting on literal `0`.

Comment (oli-obk) [inline clippy_lints/src/ranges.rs]: fixed

File: CHANGELOG.md (hunk 1 above)
Comment (oli-obk) [inline CHANGELOG.md]: fixed

**Pull request merged**